### PR TITLE
docs: Add docs around buffering.

### PR DESCRIPTION
**Pull request by amanda-tarafa**
Towards #7636.

### CODE REVIEW

File: apis/Google.Cloud.Diagnostics.AspNetCore/docs/index.md [hunk 1]
```diff
@@ -36,11 +36,14 @@ for the permissions needed for Tracing.
...
+> When running on environments with limited CPU for background activities, take care not to use the
 > timed buffer options for any of Logging, Tracing or Error Reporting. Take into account that the timed buffer
-> is used for all of these components by default so you will need to explicitly configure the buffers by using
-> the `Google.Cloud.Diagnostics.Common.LoggerOptions`, `Google.Cloud.Diagnostics.Common.TraceOptions` and
-> `Google.Cloud.Diagnostics.Common.ErrorReportingOptions` classes.
+> is used for the Logging and Tracing components by default so you will need to explicitly configure the buffers by using
+> the `Google.Cloud.Diagnostics.Common.LoggerOptions` and `Google.Cloud.Diagnostics.Common.TraceOptions` classes. The Error Reporting
+> component does not buffer entries by default.
+> See the
+> [`Google.Cloud.Diagnostics.Common`](https://cloud.google.com/dotnet/docs/reference/Google.Cloud.Diagnostics.Common/latest)
```
Comment (jskeet) [inline apis/Google.Cloud.Diagnostics.AspNetCore/docs/index.md:45]: Move the word "documentation" into the link?

File: apis/Google.Cloud.Diagnostics.AspNetCore3/docs/index.md [hunk 2]
```diff
@@ -36,11 +36,14 @@ for the permissions needed for Tracing.
...
+> When running on environments with limited CPU for background activities, take care not to use the
 > timed buffer options for any of Logging, Tracing or Error Reporting. Take into account that the timed buffer
-> is used for all of these components by default so you will need to explicitly configure the buffers by using
-> the `Google.Cloud.Diagnostics.Common.LoggerOptions`, `Google.Cloud.Diagnostics.Common.TraceOptions` and
-> `Google.Cloud.Diagnostics.Common.ErrorReportingOptions` classes.
+> is used for the Logging and Tracing components by default so you will need to explicitly configure the buffers by using
+> the `Google.Cloud.Diagnostics.Common.LoggerOptions` and `Google.Cloud.Diagnostics.Common.TraceOptions` classes. The Error Reporting
+> component does not buffer entries by default.
+> See the
+> [`Google.Cloud.Diagnostics.Common`](https://cloud.google.com/dotnet/docs/reference/Google.Cloud.Diagnostics.Common/latest)
```
Comment (jskeet) [inline apis/Google.Cloud.Diagnostics.AspNetCore3/docs/index.md:45]: And again.

File: apis/Google.Cloud.Diagnostics.Common/docs/index.md [hunk 3]
```diff
@@ -70,6 +70,37 @@ This can be useful if you only need to use some of the observability components.
...
+
+These strategies are:
+
+- Timed buffers: Written entries are stored in a buffer that is flushed periodically to the backend service.
+  This is the buffering strategy used by default by the Tracing and Logging components. Since the service calls
+  are performed in the background, this buffering strategy won't affect your application's performance, in particular
+  it won't affect external requests or user interactions. Depending on the amount of entries your application is generating, you may
+  want to fine tune the timed buffer period, which defaults to 5 seconds, to flush more or less frequently by taking into account
+  application memory comsumption and how soon you need entries to appear on the backend.
+  As stated earlier on this documentation, it is important to note that some environments do limit CPU usage for background
```
Comment (jskeet) [inline apis/Google.Cloud.Diagnostics.Common/docs/index.md:88]: on this => in this

File: apis/Google.Cloud.Diagnostics.Common/docs/index.md [hunk 4]
```diff
@@ -70,6 +70,37 @@ This can be useful if you only need to use some of the observability components.
...
+  operations, and in such cases, using timed buffers may result on delayed backend entry recording, lost entries or even thread starvation
```
Comment (jskeet) [inline apis/Google.Cloud.Diagnostics.Common/docs/index.md:89]: result on => result in

File: apis/Google.Cloud.Diagnostics.Common/docs/index.md [hunk 5]
```diff
@@ -70,6 +70,37 @@ This can be useful if you only need to use some of the observability components.
...
+  scenarios.
+- Sized buffers: Written entries are stored in a buffer that is flushed to the backend service when it reaches a given size.
+  Service calls are not performed in the background, so the sized buffer may affect performance of certain external requests or
+  user interactions, in particular those who make the buffer reach the flushing size. In general you'll need to fine tune the
+  buffer size so that flushing occurs as little as possible while keeping memory comsumption at appropiate levels. Adjustments will
```
Comment (jskeet) [inline apis/Google.Cloud.Diagnostics.Common/docs/index.md:94]: appropiate => appropriate

**Review by jskeet — APPROVED** [review]
Generally fine - just a few nits

File: apis/Google.Cloud.Diagnostics.AspNetCore/docs/index.md (hunk 1 above)
Comment (amanda-tarafa) [reply apis/Google.Cloud.Diagnostics.AspNetCore/docs/index.md:45]: Done, here and in the other one.

File: apis/Google.Cloud.Diagnostics.Common/docs/index.md (hunk 3 above)
Comment (amanda-tarafa) [reply apis/Google.Cloud.Diagnostics.Common/docs/index.md:88]: Done

File: apis/Google.Cloud.Diagnostics.Common/docs/index.md (hunk 4 above)
Comment (amanda-tarafa) [reply apis/Google.Cloud.Diagnostics.Common/docs/index.md:89]: Done

File: apis/Google.Cloud.Diagnostics.Common/docs/index.md (hunk 5 above)
Comment (amanda-tarafa) [reply apis/Google.Cloud.Diagnostics.Common/docs/index.md:94]: done

**Review by amanda-tarafa — COMMENTED** [review]
All nits addressed, will push shortly and merge on green.

**Pull request merged**